environment/ssh.rst

```diff
@@ -71,89 +78,248 @@ this::
+  instructions carefully to avoid overwriting your existing SSH key pair and
+  losing access to GitHub.
+
+
+**Step 1**: Open a terminal window on your local computer. On Windows, open a
+Powershell terminal by pressing the Windows button and searching for
+“Powershell”. This terminal should be on your **local machine**; that is, **do
+not connect with SSH to the Linux server**. The following commands (steps 2-5)
+should be run on your local machine.
+
+
+**Step 2**: Run ``$ cd``
+
+Remember that $ is used to denote a terminal command, it is not part of the
+command itself. This command will take you to your home directory. 
+
+**Step 3**: Run ``$ ssh-keygen``
+
+You will see the following prompt: ::
+
+  Generating public/private rsa key pair. 
+  Enter file in which to save the key (/home/username/.ssh/id_rsa):
+
+Press Enter to accept the default location. Check the output of this command to
+decide whether you need to generate a new SSH key pair (see the warning below).
+
+.. warning :: 
+  If you have already completed generated an SSH key pair for (possibly for 
+  GitHub) , you will see a prompt like this:
+  ::
+
+    /home/username/.ssh/id_rsa already exists. Overwrite (y/n)?
+
+  If you see this prompt, **do not overwrite your existing SSH key pair**. You
+  can re-use this key pair. **Skip ahead to Step 4**.
+
+You will then see the following prompt:
+
+::
+
+  Enter passphrase (empty for no passphrase):
+
+In order to use SSH without a password, you must leave this passphrase empty.
+Press Enter to accept the default (empty passphrase). You will then see a
+message like this:
+
+::
+
+  Your identification has been saved in /home/username/.ssh/id_rsa. 
+  Your public key has been saved in /home/username/.ssh/id_rsa.pub. 
+  The key fingerprint is:
+  SHA256:cBUUs2FeMCIrBlTyv/PGpBtNz0v235zvLykpoWIOS9I username@machine
+
+  The key’s randomart image is:
+  +---[RSA 3072]----+
+  | .+.. . ..@+.    |
+  | + o = *         |
+  | + o . o         |
+  | . o o           |
+  | . S             |
+  | . +.o.          |
+  | . E ++..=. . .  |
+  | o o+++o.oo oo.  |
+  | .oo+. ...o.+O   |
+  +----[SHA256]-----+
+
+This message indicates that your SSH key pair has been generated successfully.
+
+**Step 4**: Time to copy your public key to the Linux server. This command will
+differ depending on whether you are using a Mac or Windows computer.
+
+- **Mac/Linux**: Run ``$ ssh-copy-id CNETID@<servername>.cs.uchicago.edu``
+  
+    Replace ``CNETID`` with your CNetID, and ``<servername>`` with the name of
+    the server you want to connect to. For example, if you want to connect to
+    such as ``linux1.cs.uchicago.edu`` or ``cs141-4.cs.uchicago.edu``.
+
+    You will be prompted for your password. Enter your CNetID password.
+
+- **Windows**: Run the following command:
+
+  ``$ type $env:USERPROFILE\\.ssh\\id_rsa.pub | ssh
+  CNETID@<servername>.cs.uchicago.edu “cat >> .ssh/authorized_keys”``
```

Comment:
Use a code block (with three back-ticks) to ensure this long command is formatted better.